sogs/model/bot.py

```diff
@@ -0,0 +1,267 @@
+from __future__ import annotations
+
+from .. import crypto, db, config
+from ..db import query
+from ..web import app
+from .exc import BadPermission, PostRateLimited
+from .. import utils
+from ..omq import omq_global
+from .user import User
+from .room import Room
+from .message import Message
+from .filter import SimpleFilter
+from .exc import InvalidData
+
+from typing import Optional, List, Union
+import time
+
+
+class Bot:
+    """
+    Class representing a simple bot to manage open group server
+
+    Object Properties:
+        id - database primary key for user row
+        status - bot active(T)/passive(F) moderation status (key = room, value = bool)
+        rooms - reference to room(s) in which bot is active
+        filter - reference to filter object paired with bot
+        perm_cache - dict storing user info/status (synced from all rooms patrolled)
+        session_id - hex encoded session_id of the bot
+        banned - default to false
+        global_admin - default to true for bot
+        global_moderator - default to true for bot
+        visible_mod - default to true for bot
+    """
+
+    def __init__(
+        self,
+        _rooms: List[Room],
+        row=None,
+        *,
+        id: Optional[int] = None,
+        session_id: Optional[int] = None,
+    ) -> None:
+        # immutable attributes
+        self._banned: bool = False
+        self._global_admin: bool = False
+        self._global_moderator: bool = False
+        self._visible_mod: bool = False
+        self.id = id
+
+        # operational attributes
+        self.rooms: List[Room] = _rooms
+        self.status = {r: False for r in self.rooms}
+        self.filter = SimpleFilter(_bot=self)
+        self.perm_cache = {}
+        self.current_message: Message = None
+        self.nlp_model = None
+        self.language = 'English'
+        self.word_blacklist = ['placeholderlist', 'until', 'we', 'decide', 'naughty', 'words']
+
+    def __setattr__(self, __name: str, __value) -> None:
```

Comment:
why are we hooking `__setattr__` and `__getattr__` here?

the norm is usually that all `_` prefixed members are considered invisible to other types.

Reply:
this class is not going to exist soon